tests/system/src/test/java/test/robot/helloworld/RectangleTest.java

```diff
@@ -159,4 +164,26 @@ public void testAddFillRect() {
         });
     }
 
+    @Test(timeout = 15000)
+    public void testScreenCaptureRect() {
+        final int WIDTH = 400;
+        final int HEIGHT = 300;
+        runAndWait(() -> {
+            Group root = new Group();
+            testScene = new Scene(root, WIDTH, HEIGHT);
+            testScene.setFill(Color.PALEGREEN);
+
+            testStage = getStage();
+            testStage.setScene(testScene);
+            testStage.show();
+        });
+        waitFirstFrame();
+        runAndWait(() -> {
+            Robot robot = new Robot();
+            WritableImage capture = robot.getScreenCapture(null, 0, 0, WIDTH, HEIGHT);
+            assertEquals(capture.getWidth(), WIDTH);
+            assertEquals(capture.getHeight(), HEIGHT);
```

Comment:
You need to use the 3-arg version of assertEquals with some small delta value. As it is, I get the following test failure:

test.robot.helloworld.RectangleTest > testScreenCaptureRect FAILED
    java.lang.AssertionError: Use assertEquals(expected, actual, delta) to compare floating-point numbers
        at org.junit.Assert.fail(Assert.java:91)
        at org.junit.Assert.assertEquals(Assert.java:491)
        at org.junit.Assert.assertEquals(Assert.java:480)
        at test.robot.helloworld.RectangleTest.lambda$testScreenCaptureRect$11(RectangleTest.java:184)

Reply:
also, I just noticed that the expected and actual values are backwards. The first arg should be the expected value.